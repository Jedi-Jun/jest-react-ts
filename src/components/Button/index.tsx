import styled from 'styled-components';

interface ContainerProps {
  readonly backgroundColor?: string;
  readonly hoverColor?: string;
}

const ButtonStyle = styled.button<ContainerProps>`
  background: ${(props) => props.backgroundColor || '#304ffe'};
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  border: 0;
  &: hover {
    background: ${(props) => props.hoverColor || '#1e40ff'};
  }
  &: active {
    box-shadow: inset 10px 10px 20px rgba(0, 0, 0, 0.2);
  }
`;
const Label = styled.div`
  color: #ffffff;
  font-size: 16px;
`;

interface Props {
  readonly label: string;
  readonly backgroundColor?: string;
  readonly hoverColor?: string;
  readonly onClick?: () => void;
}

export const Button = ({ onClick, label, backgroundColor, hoverColor }: Props) => {
  return (
    <ButtonStyle onClick={onClick} backgroundColor={backgroundColor} hoverColor={hoverColor}>
      <Label>{label}</Label>
    </ButtonStyle>
  );
};
