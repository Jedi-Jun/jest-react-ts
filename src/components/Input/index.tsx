import React from 'react';
import styled from 'styled-components';

const InputStyle = styled.input.attrs({
  type: 'text',
})`
  display: flex;
  font-size: 16px;
  padding: 8px 10px;
  border-radius: 8px;
  border: 3px solid #bdbdbd;
  outline: none;
`;

interface Props {
  readonly value?: string;
  readonly onChange?: (text: string) => void;
  readonly placeholder?: string;
  readonly onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
}

export const Input = ({ value, onChange, placeholder, onKeyPress }: Props) => {
  const handlerChange = (event) => {
    if (typeof onChange === 'function') {
      onChange(event.target.value);
    }
  };
  return (
    <InputStyle
      value={value}
      placeholder={placeholder}
      onChange={handlerChange}
      onKeyPress={onKeyPress}
    />
  );
};
