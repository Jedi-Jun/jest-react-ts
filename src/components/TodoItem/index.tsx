import styled from 'styled-components';
import { Button } from 'components';

const Container = styled.div`
  display: flex;
  border: 1px solid blue;
  align-items: center;
  justify-contents: center;
  // align-items: bottom;
  // padding: 10px;
`;

const Label = styled.label`
  flex: 1;
  font-size: 1.3rem;
  margin-right: 10px;
  border-top: 1px solid blue;
`;

interface Props {
  readonly label: string;
  readonly onDelete?: () => void;
}

export const ToDoItem = ({ label, onDelete }: Props) => {
  return (
    <Container>
      <Label>{label}</Label>
      <Button label="Del" backgroundColor="#ff1744" hoverColor="#d01440" onClick={onDelete} />
    </Container>
  );
};
