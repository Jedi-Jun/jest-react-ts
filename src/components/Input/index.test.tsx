import { render, screen, fireEvent } from '@testing-library/react';
// render, screen, fireEvent
import { Input } from 'components';
import 'jest-styled-components';

describe('<Input />', () => {
  // #1
  it('renders component correctly', () => {
    const { container } = render(<Input value="default value" />);
    const input = screen.getByDisplayValue('default value');
    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  // #2
  it('renders placeholder correctly', () => {
    render(<Input placeholder="default placeholder" />);
    const input = screen.getByPlaceholderText('default placeholder');
    expect(input).toBeInTheDocument();
  });
  // #3
  it('changes the data', () => {
    render(<Input placeholder="default placeholder" />);
    const input = screen.getByPlaceholderText('default placeholder') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'Hello Jest' } });
    expect(input.value).toBe('Hello Jest');
  });
});
