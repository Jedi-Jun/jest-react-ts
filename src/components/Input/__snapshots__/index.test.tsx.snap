// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders component correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  padding: 8px 10px;
  border-radius: 8px;
  border: 3px solid #bdbdbd;
  outline: none;
}

<div>
  <input
    class="c0"
    type="text"
    value="default value"
  />
</div>
`;
